post_processing/exporters/opensearch_exporter.py

```diff
@@ -0,0 +1,586 @@
+            auth_token: Optional bearer token for authentication
+            username: Optional username for basic auth
+            password: Optional password for basic auth
+            verify_ssl: Whether to verify SSL certificates
+            timeout: Request timeout in seconds
+            max_retries: Maximum number of retry attempts
+            retry_delay: Delay between retries in seconds
+        """
+        self.url = url.rstrip('/')
+        self.index = index
+        self.auth_token = auth_token
+        self.username = username
+        self.password = password
+        self.verify_ssl = verify_ssl
+        self.timeout = timeout
+        self.max_retries = max_retries
+        self.retry_delay = retry_delay
+        self.logger = logging.getLogger(__name__)
+
+    def _build_headers(self) -> Dict[str, str]:
+        """Build HTTP headers for requests."""
+        headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json'
+        }
+
+        if self.auth_token:
+            headers['Authorization'] = f'Bearer {self.auth_token}'
+        elif self.username and self.password:
+            # Basic auth
+            import base64
+            credentials = f"{self.username}:{self.password}"
+            encoded = base64.b64encode(credentials.encode('utf-8')).decode('ascii')
+            headers['Authorization'] = f'Basic {encoded}'
+
+        return headers
+
+    def _make_request(
+        self,
+        endpoint: str,
+        method: str = 'POST',
+        data: Optional[Dict] = None,
+        is_bulk: bool = False
+    ) -> Dict[str, Any]:
+        """
+        Make HTTP request to OpenSearch with retry logic.
+
+        Args:
+            endpoint: API endpoint path
+            method: HTTP method
+            data: Request payload (Dict for regular requests, str for bulk)
+            is_bulk: Whether this is a bulk request (uses different content-type)
+
+        Returns:
+            Response data as dictionary
+
+        Raises:
+            Exception: If request fails after retries
+        """
+        url = urljoin(self.url, endpoint)
+        headers = self._build_headers()
+
+        # For bulk requests, use different content-type
+        if is_bulk:
+            headers['Content-Type'] = 'application/x-ndjson'
+
+        for attempt in range(self.max_retries):
+            try:
+                # Prepare request
+                if is_bulk:
+                    # Bulk data is already a string
+                    request_data = data.encode('utf-8') if data else None
+                else:
+                    # Regular JSON data
+                    request_data = json.dumps(data).encode('utf-8') if data else None
+
+                req = urllib.request.Request(
+                    url,
+                    data=request_data,
+                    headers=headers,
+                    method=method
+                )
```

Comment:
Isn't there an opensearch module for python?

We could stay with `requests` to minimize dependencies.  @frival @dvalinrh thoughts?